src/stactools/naip/stac.py

```diff
@@ -121,30 +124,51 @@ def create_item(
         )
 
     if fgdc_metadata_href is not None:
-        fgdc_metadata_text = read_text(fgdc_metadata_href)
-        fgdc = parse_fgdc_metadata(fgdc_metadata_text)
-    else:
-        fgdc = {}
+        if year == "2020":
+            file_xpath = """gmd:identificationInfo/gmd:MD_DataIdentification/gmd:citation/\n
+            gmd:CI_Citation/gmd:title/gco:CharacterString"""
+
+            with fsspec.open(fgdc_metadata_href) as file:
+                root = XmlElement(
+                    etree.parse(file, base_url=fgdc_metadata_href).getroot()
+                )
+                id = root.find_text_or_throw(
+                    "".join(file_xpath.split()), missing_element("File Identifier")
+                )
+                dt = str_to_datetime(id.split(".")[0].split("_")[-1])
+                if id is None:
+                    id = os.path.basename(cog_href)
+                if dt is None:
+                    fname = os.path.splitext(os.path.basename(cog_href))[0]
+                    fname_date = fname.split("_")[5]
```

Comment:
this should use a regex rather than splitting a 6 item field and be more defensive if the string isn't the expected format.

Also, this logic is used in both the text and xml branches, so they should both use a common method. One approach I usually use is for this code to pull the needed values out of the metadata file in the branch, and then have both call a method that parses those values.

Reply:
Updated to use regex. 

This logic is used to determine the 1) file name and 2) date, from the COG file's name, if either no metadata file exists for a scene, or the required fields from which to extract these two items don't exist in the metadata file. Some of the scenes have either no metadata files (those from 2010), or some of them have metadata files but the above information cannot be found within those metadata files (for e.g. the key 'Distribution Information' might not exist in some of the TXT metadata files). It is only for these files that this method is called and the date and file name are extracted by using the COG href instead of the FGDC metadata href.